src/main/java/com/walking/intensive/chapter2/task8/Task8.java

```diff
@@ -1,30 +1,39 @@
 package com.walking.intensive.chapter2.task8;
 
-/**
- * Добираясь в школу на трамвае Вова проверяет, является ли купленный билет счастливым.
- * Если является, то нужно загадать желание и съесть билетик.
- *
- * <p>Билет содержит 6 цифр, комбинации подбираются случайным образом от 000000 до 999999.
- *
- * <p>Билет считается счастливым, если сумма первых трех цифр равна сумме последних трех цифр.
- *
- * <p>Пример: 123411 – счастливый (1 + 2 + 3 == 4 + 1 + 1).
- *
- * <p>Реализуйте метод getHappyTicketChance(), который будет возвращать вероятность
- * выпадения счастливого билета.
- *
- * <p>P.S. Вероятность – это отношение благоприятных исходов к числу всех исходов.
- * Вероятность не может принимать значение больше 1.
- *
- * <p><a href="https://github.com/KFalcon2022/intensive-tasks-2024/blob/master/README.md">Требования к оформлению</a>
- */
+import java.util.Arrays;
+
 public class Task8 {
     public static void main(String[] args) {
-//        Для собственных проверок можете делать любые изменения в этом методе
+
+        System.out.println(getHappyTicketChance());
     }
 
     static double getHappyTicketChance() {
-        // Ваш код
-        return 0.0;
+
+        int[] sumOfDigits = new int[28];
+        int currentDigit;
+        int summ;
+
+        for (int i = 1; i < 1000; i++) {
+            currentDigit = i;
+            summ = 0;
+
+            while (currentDigit > 0) {
+                summ += currentDigit % 10;
+                currentDigit /= 10;
+            }
+
+            sumOfDigits[summ]++;
+        }
+
+        sumOfDigits[0] = 1;                 //Вероятность выпадения комбинации 000 = 1/1000
+
+        double totalChance = 0;
+
+        for (int i = 0; i < 28; i++) {
+            totalChance += Math.pow(sumOfDigits[i], 2);
+        }
+
+        return totalChance/1000000;
```

Comment:
Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter2/task8/Task8.java:37:27: '/' is not followed by whitespace. [WhitespaceAround]
Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter2/task8/Task8.java:37:[27](https://github.com/KFalcon2022/intensive-tasks-2024/actions/runs/12031936068/job/33578641940?pr=385#step:4:28): '/' is not preceded with whitespace. [WhitespaceAround]